sdk/textanalytics/ai-text-analytics/src/azureKeyCredentialPolicy.ts

```diff
@@ -0,0 +1,49 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT license.
+
+import { KeyCredential } from "@azure/core-auth";
+import {
+  RequestPolicyFactory,
+  RequestPolicy,
+  RequestPolicyOptions,
+  BaseRequestPolicy,
+  WebResource
+} from "@azure/core-http";
+
+const API_KEY_HEADER_NAME = "Ocp-Apim-Subscription-Key";
+
+/**
+ * Create an HTTP pipeline policy to authenticate a request
+ * using an `AzureKeyCredential` for Text Analytics
+ */
+export function createTextAnalyticsAzureKeyCredentialPolicy(
```

Comment:
thoughts about making a general `KeyCredentialPolicy` in `core-http` that takes the header name in its create factory?

Reply:
@bterlson and I were discussing this earlier. There are some open questions about whether or not we should use a code-sharing approach (what .NET does) or actually stabilize the policy in core-http's API surface.

Reply:
It may be that simply setting a header to a key is common enough that we can just have it in core, as a regular thing. @maririos can you share some thoughts on why .net went with the code sharing approach?

Reply:
>simply setting a header to a key is common enough

True. We expect most of the libraries to just use this simple approach. In case there is a scenario that requires more customization or more complex logic, then the library can implement its own policy, therefore opt in or out of the private shared policy.

Reply:
@bterlson @xirzec similar to the generic type parameter, this is something we could move to in the future. As it stands, this PR only puts the credential types into the API surface of both text analytics and core-auth, but the policy is completely internal. I think we could merge this as it is now and take a work item to either move the policy to a shared source like in .NET or decide if we want to stabilize it through core-http.